### ISSUE TITLE
Move modules from mod/ to src/Module

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#4090]
In the same vein as #3878, I laid the ground work for moving files in the `mod/` folder to the `Friendica\Module` namespace.

The new Module classes must extend `BaseModule` and the module functions are now the following:
- `init()`
- `content()`
- `post()`
- `afterpost()`

There isn't any `App` parameter anymore, the App object can be retrived via `self::getApp()` as `BaseModule` extends `BaseObject` as well.

Modules can have additional static functions beyond the base functions outlined above, however please consider if those functions should be in another class, especially if it is shared by multiple modules.

**Comment by annando** [comment]
I would add that if there are more methods, then only private methods.

**Comment by MrPetovan** [comment]
In the `Login` module case, we need to be able to show a login form in any other module, which means the `form()` method of the `Login` module has to be public, so I'm not favorable to this strict rule, but if the method is only used in the same module, it should be private indeed.

**Comment by nupplaphil** [comment]
@MrPetovan 
Because @JonnyTischbein already started moving the next part, I share my thoughts a little bit earlier than I wanted (nevertheless we should merge his PR before changing something I'm now explaining ^^).

I'm thinking about refactoring the way we use the `Friendica\BaseModule` currently.
- Moving the whole Module logic out of `index.php` to the `Friendica\App` class
- Therefore changing the `Friendica\BaseModule` from a static class to a "real" instance which is hold inside `App`.
- So we can inject the necessary dependencies with the help of the method signature rather than calling static methods inside.
- I'm thinking (as you do) about a routing class which will use the module instances (but there are just some vague thoughts ^^)
- (of course another reason is that we can mock it better for tests ;-) )

Is there a particular reason that you created it all statically?
What do you say in general about it?

**Comment by MrPetovan** [comment]
> Is there a particular reason that you created it all statically?

Yes, it was the first necessary step to clear the `mod/` folder without rocking the boat too much. The router is the next logical step to split `mod/api.php` into sub-modules than can be registered against multiple routes. Adding dependency injection is a nice-to-have feature that can take place after `mod/` is definitively deleted.

**Comment by nupplaphil** [comment]
All clear :-)

So I move my focus back to `boot.php` ;-)

**Comment by annando** [comment]
I think the whole philosophy behind this could be called "don't make it too complicated". Having a file for every path improves debugging a lot. See the new files I created for ActivityPub as an example. The modules "Objects", "Inbox", "Outbox", "Following" and "Followers" are very small and easily understandable.

**Comment by annando** [comment]
Please leave `mod/item.php` until the end. There is too much functionality in `item_post`. Plan is to move everything needed to new functions and to remove redundant stuff.
